### PR TITLE
Remind me later

### LINKED ISSUE #1313: Notify me later about this message
- Overview ticket at https://github.com/nextcloud/spreed/issues/10075

---

The message menu should have an option:
* "Set reminder" > Submenu opens with following options
grafik

## Required aka. MVP
- [x] Later today => 6pm
- [x] Tomorrow => 8am
- [x] This weekend => 8am Saturday
- [x] Next week => 8am Monday

## Optional

- [x] Custom date+time picker

### PR DESCRIPTION
**Pull request by SystemKeeper**
Fixes #1313 

Notes:
* On the screenshot it says 8am in all cases, but for "Later today" it's 6pm as specified and is correct in code
* Wording might change, waiting for the decision on #1313 about that, but did go with "Next Saturday" and "Next Monday" for now (CC @nimishavijay for the current state)
* When "Today" is already >= 6pm, the option is hidden
* For custom date & time the minimum selectable time would be 15 min in the future (so no reminders in the past)

### CODE REVIEW

File: NextcloudTalk/en.lproj/Localizable.strings [hunk 1]
```diff
@@ -1267,6 +1282,9 @@
 /* No comment provided by engineer. */
 "Reference API supported?" = "Reference API supported?";
 
+/* Remind me later about that message */
+"Remind me later" = "Remind me later";
```
Comment (nickvergessen) [inline NextcloudTalk/en.lproj/Localizable.strings]:
```suggestion
"Set reminder" = "Set reminder";
```

File: NextcloudTalk/NCChatViewController.m [hunk 2]
```diff
@@ -4264,6 +4271,98 @@ - (UIContextMenuConfiguration *)tableView:(UITableView *)tableView contextMenuCo
...
+
+        NSArray *subActions;
+
+        if (now.hour >= 18) {
+            subActions = @[tomorrow, nextSaturday, nextMonday, customReminder];
+        } else {
+            subActions = @[laterToday, tomorrow, nextSaturday, nextMonday, customReminder];
+        }
+
+        UIMenu *remindeMeLaterMenu = [UIMenu menuWithTitle:NSLocalizedString(@"Remind me later", @"Remind me later about that message")
```
Comment (nickvergessen) [inline NextcloudTalk/NCChatViewController.m]:
```suggestion
        UIMenu *remindeMeLaterMenu = [UIMenu menuWithTitle:NSLocalizedString(@"Set remind", @"Remind me later about that message")
```

**Review by nickvergessen — CHANGES_REQUESTED** [review]
Ref https://github.com/nextcloud/server/issues/39181#issuecomment-1662059862

**Comment by SystemKeeper** [comment]
> Ref [nextcloud/server#39181 (comment)](https://github.com/nextcloud/server/issues/39181#issuecomment-1662059862)

Done, updated screenshot:

**Comment by SystemKeeper** [comment]
Adjusted options as discussed:


Delete option if a reminder is already available:

**Comment by nickvergessen** [comment]
At least on the backend you could still set the reminder (it gets updated/overwritten) as well as deleting it.
According to https://github.com/nextcloud/server/issues/39181#issuecomment-1664832347 the option would be called "Clear reminder" and below that there are the normal options to set the reminder to a different value.

**Comment by SystemKeeper** [comment]
Moved to inline menu:


@nickvergessen ✅?

**Review by nickvergessen — APPROVED** [review]
swift code looks good, other code I can not understand, but seems to do it's job and screenshot is 🤩

**Pull request merged**